src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,227 @@
+package duke;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+
+public class Parser {
+    private TaskList taskList;
+    /** An ArrayList of Tasks obtained from TaskList **/
+    private ArrayList<Task> tasks;
+
+    /** A collection of commands Tasker can execute **/
+    public enum Command {
+        LIST,
+        DONE,
+        DELETE,
+        TODO,
+        DEADLINE,
+        EVENT,
+        FIND
+    }
+
+    /**
+     * Constructs a Parser object.
+     * @param taskList TaskList
+     */
+    public Parser(TaskList taskList) {
+        this.taskList = taskList;
+        this.tasks = this.taskList.getTasks();
+    }
+
+    /**
+     * Generates output in response to the user's command.
+     * @param command Command in the form of a String array
+     * @throws DukeException If command could not be understood by Tasker.
+     */
+    public void receive(String[] command) throws DukeException {
+        try {
+            String task = command[0].toUpperCase();
+            Command c = Command.valueOf(task);
+            switch (c) {
+            case LIST:
+                listTasks(tasks);
+                break;
+            case DONE:
+                int index = Integer.parseInt(command[1]);
+                markTaskDone(tasks, index);
+                break;
+            case DELETE:
+                int i = Integer.parseInt(command[1]);
+                deleteTask(tasks, i);
+                break;
+            case TODO:
+                createTodo(tasks, command);
+                break;
+            case DEADLINE:
+                createDeadline(tasks, command);
+                break;
+            case EVENT:
+                createEvent(tasks, command);
+                break;
+            case FIND:
+                String keyword = command[1];
+                search(tasks, keyword);
+                break;
+            default:
+                throw(new IllegalArgumentException());
+            }
+        } catch (IllegalArgumentException e) {
+            throw new DukeException("Please enter a legit command...");
+        }
+    }
+
+    public ArrayList<Task> consolidate() {
+        return tasks;
+    }
+
+    /**
+     * Lists all the tasks in the TaskList.
+     * @param tasks An ArrayList of Tasks
+     */
+    public static void search(ArrayList<Task> tasks, String keyword) {
+        System.out.println("----------------------------------------------");
+        System.out.println("Here are the matching tasks in your list:");
+        for (int i = 0; i < tasks.size(); i++) {
+            Task task = tasks.get(i);
+            if (haveWord(task.getDescription(), keyword)) {
+                System.out.println((i + 1) + ". " + tasks.get(i));
+            }
+        }
+        System.out.println("----------------------------------------------");
+    }
+
+    public static boolean haveWord(String description, String keyword) {
```

Comment:
Should haveWord be hasWord instead? This way it would sound more like a boolean method.